numpy/core/src/multiarray/convert_datatype.c

```diff
@@ -2081,14 +2084,15 @@ PyArray_ConvertToCommonType(PyObject *op, int *retn)
             newtype = PyArray_DescrFromObject(otmp, stype);
             Py_XDECREF(stype);
```

Comment:
Same again - you're done with the object already by this line, so can decref it immediately

Reply:
done